lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -889,6 +889,18 @@ void PatchEntryPointMutate::setFuncAttrs(Function *entryPoint) {
     spiPsInputAddr.bits.sampleCoverageEna = builtInUsage.sampleMaskIn;
 
     builder.addAttribute("InitialPSInputAddr", std::to_string(spiPsInputAddr.u32All));
+
+    unsigned colFormat = m_pipelineState->getPalMetadata()->getRegister(mmSPI_SHADER_COL_FORMAT);
```

Comment:
How do we handle the separate compilation case? For example, you can use amdllpc to compile `llpc/test/shaderdb/OpKill_TestGeneral_lit.frag`, I guess the colFormat will always be zero even we output color in PS?

Reply:
Good point! However, I think we can infer that there are exports because the function returns.
In the separate compilation who fixes up the value of SPI_SHADER_COL_FORMAT? PAL?

Reply:
In a separate shader compilation, the same LGC code to add the SPI_SHADER_COL_FORMAT entry to PAL metadata is used, but it is called from colour export glue shader generation at link time. So you won't see it here. Does that mean that your case will be slightly suboptimal in the separate shader compilation case?

Reply:
I think we can refine the attribute to something like `hasColorExport` and `hasDepthExport`, this exactly match what we need to know in the backend. We have gathered this information in `FragColorExport.cpp`, so I think we should have a way to get that information and set the attributes accordingly even for separate compilation. The exact format is unknown until we generate the export shader for separate compilation case.

Reply:
I have reworked this so it also uses `getPalMetadata()->getColorExportCount()` which should cover the separate compilation case.

Reply:
looks good to me. But I think we need to wait https://reviews.llvm.org/D105683 get landed?

Reply:
This can be merged without backend support.  Once this and backend support are in, then I will follow up with changes that do required backend support.